### PR TITLE
6-GET /api/articles/:article_id/comments (UPDATED)

### PR DESCRIPTION
**Pull request by booayl**
created new endpoint of: /api/articles/:article_id/comments

Happy Path:
GET200 : responds with an array of comments objects from selected article by article ID. Comments served with the most recent comments first.

Sad Path:
GET404: Non-existent ID
GET400: Invalid Input Type

*Refactored after reading feedback from mentors:
1. changed typeof to .toMatchObject in test suites
2. changed "400 Invalid Input ID" to "404 Non-existent ID"

### CODE REVIEW

File: __tests__/seed.test.js [hunk 1]
```diff
@@ -114,31 +119,33 @@ describe("GET /api/articles/:article_id/comments", () => {
...
-          expect(typeof comment.author).toBe("string");
-          expect(typeof comment.body).toBe("string");
-          expect(comment.article_id).toBe(1);
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: 1,
```
Comment (roisinoreilly) [inline __tests__/seed.test.js:128]: perfect

File: __tests__/seed.test.js [hunk 2]
```diff
@@ -114,31 +119,33 @@ describe("GET /api/articles/:article_id/comments", () => {
...
-          expect(typeof comment.comment_id).toBe("number");
-          expect(typeof comment.votes).toBe("number");
-          expect(typeof comment.created_at).toBe("string");
-          expect(typeof comment.author).toBe("string");
-          expect(typeof comment.body).toBe("string");
-          expect(comment.article_id).toBe(1);
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
```
Comment (roisinoreilly) [inline __tests__/seed.test.js:125]: you can also use any(Date) here for a bit of extra security

**Review by roisinoreilly — COMMENTED** [review]
excellent tests and code again, really great job. One very small pointer I have is just that I would change your test file name to index.test.js or you can separate the tests by different endpoints eg. articles.test.js, it's not so much testing the seed as it is testing your different endpoints

**Review by roisinoreilly — COMMENTED** [review]
one thing for this ticket I'd also like you to look at is what happens if an article doesn't have any comments yet and what you want to happen (I think article id 2 has no comments for testing purposes)

File: __tests__/seed.test.js (hunk 2 above)
Comment (booayl) [reply __tests__/seed.test.js:125]: I tired doing any(Date), however it came back with error, think it's something to do with the utils functions? I did ended up passing it using 'expect.toBeDateString()' :)

Comment (roisinoreilly) [reply __tests__/seed.test.js:125]: ah interesting! that works equally as good then nice catch

**Pull request merged**